lib/klaviyo/client.rb

```diff
@@ -40,10 +40,19 @@ def self.request(method, path, content_type, **kwargs)
     end
 
     def self.public_request(method, path, **kwargs)
-      check_public_api_key_exists()
-      params = build_params(kwargs)
-      url = "#{BASE_API_URL}/#{path}?#{params}"
-      res = Faraday.get(url).body
+      check_public_api_key_is_valid()
+      if method == HTTP_GET
```

Comment:
won't this fail if someone enters `'GET'` instead of `'get'`? would it make sense to make `method` always lowercase or use something like this method I found in the ruby API docs? https://ruby-doc.org/core-1.9.3/String.html#method-i-casecmp

Reply:
While testing, it didn't cause an issue for me, tried all lowercase, camelcase and uppercase

Reply:
@siddwarkhedkar are you saying you tried it using casecmp like Nate suggested, or without that addition?  In my tests I'm seeing that using 'Get' or 'POST' are causing it to fail

Reply:
I tried with the current code on the branch, not using casecamp